### ISSUE TITLE
Install torch (GPU)

### ISSUE THREAD

**Issue by romainGuiet** [weigertlab/tarrow#4]
Dear @bentaculum , Dear @maweigert ,

I went through some troubles with the installation (with torch GPU) and I thought it could be helpful to others, please find below more information.

I created and install using the recommended lines : 

```
>mamba create --name tarrow python=3.8
>mamba activate tarrow
>pip install git+https://github.com/weigertlab/tarrow.git
```
and tested for training :
`> python train.py -c scripts/hela.yaml
`
but I got a GPU error : 
```
train.py: Run name `hela_backbone_unet`
Traceback (most recent call last):
  File "train.py", line 453, in <module>
    main(args)
  File "train.py", line 332, in main
    device, n_gpus = tarrow.utils.set_device(args.gpu)
  File "C:\Users\guiet\AppData\Local\mambaforge\envs\tarrow\lib\site-packages\tarrow\utils\set_device.py", line 65, in set_device
    raise ValueError(f"GPUs {x} not available.")
ValueError: GPUs 0 not available.
```

Alternatively I tested with 

```
>python
>>>import torch
>>>torch.cuda.is_available()
False
```

which confirm an issue with torch-GPU 😰

Therefore I uninstalled and installed back some packages using lines : 

```
pip uninstall pytorch
pip uninstall  torchvision 
mamba install pytorch torchvision pytorch-cuda=11.8 -c pytorch -c nvidia
```

following which running

```
>python
>>>import torch
>>>torch.cuda.is_available()
True 
```
So GPU issue was solved 🎉 

BUT starting the training with 
`> python train.py -c scripts/hela.yaml
`
I got the error : 
```
>python train.py -c configs/hela.yaml
Traceback (most recent call last):
  File "C:\Users\guiet\AppData\Local\mambaforge\envs\tarrow_38_2\lib\site-packages\requests\compat.py", line 11, in <module>
    import chardet
ModuleNotFoundError: No module named 'chardet'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "train.py", line 15, in <module>
    import tarrow
  File "C:\Users\guiet\AppData\Local\mambaforge\envs\tarrow_38_2\lib\site-packages\tarrow\__init__.py", line 1, in <module>
    from . import (
  File "C:\Users\guiet\AppData\Local\mambaforge\envs\tarrow_38_2\lib\site-packages\tarrow\data\__init__.py", line 1, in <module>
    from .tarrow_dataset import TarrowDataset, ConcatDatasetWithIndex
  File "C:\Users\guiet\AppData\Local\mambaforge\envs\tarrow_38_2\lib\site-packages\tarrow\data\tarrow_dataset.py", line 10, in <module>
    from torchvision import transforms
  File "C:\Users\guiet\AppData\Local\mambaforge\envs\tarrow_38_2\lib\site-packages\torchvision\__init__.py", line 6, in <module>
    from torchvision import datasets, io, models, ops, transforms, utils
  File "C:\Users\guiet\AppData\Local\mambaforge\envs\tarrow_38_2\lib\site-packages\torchvision\datasets\__init__.py", line 1, in <module>
    from ._optical_flow import FlyingChairs, FlyingThings3D, HD1K, KittiFlow, Sintel
  File "C:\Users\guiet\AppData\Local\mambaforge\envs\tarrow_38_2\lib\site-packages\torchvision\datasets\_optical_flow.py", line 13, in <module>
    from .utils import _read_pfm, verify_str_arg
  File "C:\Users\guiet\AppData\Local\mambaforge\envs\tarrow_38_2\lib\site-packages\torchvision\datasets\utils.py", line 22, in <module>
    import requests
  File "C:\Users\guiet\AppData\Local\mambaforge\envs\tarrow_38_2\lib\site-packages\requests\__init__.py", line 45, in <module>
    from .exceptions import RequestsDependencyWarning
  File "C:\Users\guiet\AppData\Local\mambaforge\envs\tarrow_38_2\lib\site-packages\requests\exceptions.py", line 9, in <module>
    from .compat import JSONDecodeError as CompatJSONDecodeError
  File "C:\Users\guiet\AppData\Local\mambaforge\envs\tarrow_38_2\lib\site-packages\requests\compat.py", line 13, in <module>
    import charset_normalizer as chardet
  File "C:\Users\guiet\AppData\Local\mambaforge\envs\tarrow_38_2\lib\site-packages\charset_normalizer\__init__.py", line 23, in <module>
    from charset_normalizer.api import from_fp, from_path, from_bytes, normalize
  File "C:\Users\guiet\AppData\Local\mambaforge\envs\tarrow_38_2\lib\site-packages\charset_normalizer\api.py", line 10, in <module>
    from charset_normalizer.md import mess_ratio
  File "charset_normalizer\md.py", line 5, in <module>
ImportError: cannot import name 'COMMON_SAFE_ASCII_CHARACTERS' from 'charset_normalizer.constant' (C:\Users\guiet\AppData\Local\mambaforge\envs\tarrow_38_2\lib\site-packages\charset_normalizer\constant.py)
```

I found a suggestion to `pip install chardet`, 
Following which I could run `>python train.py -c configs/hela.yaml` and get 

```
>python train.py -c configs/hela.yaml
train.py: Run name `hela_backbone_unet`
set_device.py: PyTorch uses GPUs 0.
train.py: Build visualisation datasets.
tarrow_dataset.py: Load images 0-92
tarrow_dataset.py: Load images 0-92
tarrow_dataset.py: Load images 0-92
tarrow_dataset.py: Load images 0-92
train.py: Build training datasets.
tarrow_dataset.py: Load images 9-92
tarrow_dataset.py: Load images 9-92
tarrow_dataset.py: Load images 9-92
tarrow_dataset.py: Load images 9-92
train.py: Build validation datasets.
tarrow_dataset.py: Load images 0-9
tarrow_dataset.py: Load images 0-9
tarrow_dataset.py: Load images 0-9
tarrow_dataset.py: Load images 0-9
train.py: Training set: 328 images
train.py: Validation set: 32 images
train.py: Number of params: 1.96 M
model.py: Write example images to tensorboard.
model.py: Write example images to tensorboard.
  0%|                                                                                                       | 0/329 [00:00<?, ?it/s]
```
🎉🥳

In summary, the lines to create a working GPU env were : 

```
mamba create --name tarrow python=3.8
mamba activate tarrow
pip install git+https://github.com/weigertlab/tarrow.git
pip uninstall pytorch
pip uninstall torchvision 
mamba install pytorch torchvision pytorch-cuda=11.8 -c pytorch -c nvidia
pip install chardet
```

Cheers,

R

Note : 
OS : Win11
GPU :  GeForce RTX2080 Ti

**Comment by romainGuiet** [comment]
Simplifided commands : 

```
>mamba create --name tarrow_38 python=3.8
>mamba activate tarrow_38
>mamba install pytorch torchvision pytorch-cuda=11.8 -c pytorch -c nvidia
>pip install git+https://github.com/weigertlab/tarrow.git
```

**Comment by bentaculum** [comment]
This is super useful, thanks!
